scripts/export_environment.py

```diff
@@ -264,19 +264,42 @@ def resolve_env(env: Environment) -> Environment:
     return dict(ResolvedEnvironment(env))
 
 
+azul_env_vars = 'AZUL_ENV_VARS'
+
+
 def export_env(env: Environment, output: Optional[TextIO]) -> None:
     """
-    Print the given environment in a form that can be evaluated by a shell.
+    Print the given environment in a form that can be evaluated by a shell,
+    unsetting variables that are no longer used.
     """
+    try:
+        old_vars = os.environ[azul_env_vars]
+    except KeyError:
+        old_vars = set()
+    else:
+        old_vars = set(old_vars.split(','))
+    assert not any(',' in var for var in env), env
```

Comment:
Is this to prevent env vars with `,` in them? 
Could this be a `reject`/`require` if an operator can specify a value that fires off this assertion?

Reply:
Good point, I'll do that.

Reply:
This is causing pep8 to break so I'm going to undo it for now :/